src/commands/deploy/deploy.ts

```diff
@@ -39,6 +41,14 @@ export async function deploy(context: IActionContext, target?: vscode.Uri | Site
         siteConfig = await target.root.client.getSiteConfig();
     }
```

Comment:
I don't think you should be calling await getDeployNodeWithTrialApp just because they have a running app.  It should only be if they deployed if there is no target.

So you can probably do something like:

```
if (target instanceof SiteTreeItem) {
  webAppSource = WebA[[Sprice/tree'
  siteConfig = await target.root.client.getSiteConfig();
} else if (!target) {
  if (ext.azureAccountTreeItem.trialAppNode) {
      target = await getDeployNodeWithTrialApp(context)
  }
}